programs/mango-v4/tests/cases/test_force_close.rs

```diff
@@ -0,0 +1,271 @@
+use super::*;
+
+#[tokio::test]
+async fn test_force_close() -> Result<(), TransportError> {
+    let test_builder = TestContextBuilder::new();
+    let context = test_builder.start_default().await;
+    let solana = &context.solana.clone();
+
+    let admin = TestKeypair::new();
+    let owner = context.users[0].key;
+    let payer = context.users[1].key;
+    let mints = &context.mints[0..4];
+    let payer_mint_accounts = &context.users[1].token_accounts[0..4];
+
+    //
+    // SETUP: Create a group and an account to fill the vaults
+    //
+
+    let mango_setup::GroupWithTokens { group, tokens, .. } = mango_setup::GroupWithTokensConfig {
+        admin,
+        payer,
+        mints: mints.to_vec(),
+        ..GroupWithTokensConfig::default()
+    }
+    .create(solana)
+    .await;
+    let collateral_token = &tokens[0];
+    let borrow_token = &tokens[1];
+
+    send_tx(
+        solana,
+        TokenEditWeights {
+            group,
+            admin,
+            mint: mints[1].pubkey,
+            init_asset_weight: 0.6,
+            maint_asset_weight: 0.8,
+            maint_liab_weight: 1.2,
+            init_liab_weight: 1.5, // changed from 1.4
+        },
+    )
+    .await
+    .unwrap();
+
+    // deposit some funds, to the vaults aren't empty
+    create_funded_account(
+        &solana,
+        group,
+        owner,
+        99,
+        &context.users[1],
+        mints,
+        100000,
+        0,
+    )
+    .await;
+
+    let liqor = send_tx(
```

Comment:
the liqor and liqee could also be created with create_funded_account more succinctly, I think?